test/integration/apiserver/apply/apply_test.go

```diff
@@ -2811,3 +2811,192 @@ spec:
+		Name("deployment").
+		Resource("deployments").
+		SubResource("scale").
+		Namespace("default").
+		Param("fieldManager", "scale_test").
+		Body([]byte(`{"spec":{"replicas": 5}}`)).
+		Do(context.TODO()).
+		Get()
+	if err != nil {
+		t.Fatalf("Error updating scale subresource: %v ", err)
+	}
+
+	deployment, err = client.AppsV1().Deployments("default").Get(context.TODO(), "deployment", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to retrieve object: %v", err)
+	}
+
+	if *deployment.Spec.Replicas != 5 {
+		t.Fatalf("Expected replicas to be 5, but got: %d", *deployment.Spec.Replicas)
+	}
+
+	assertReplicasOwnership(t, (*deployment).GetManagedFields(), "scale_test")
+
+	// Re-apply the original object, it should fail with conflict because replicas have changed
+	_, err = client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Name("deployment").
+		Param("fieldManager", "apply_test").
+		Body(validDeployment).
+		Do(context.TODO()).
+		Get()
+	if !apierrors.IsConflict(err) {
+		t.Fatalf("Expected conflict error but got: %v", err)
+	}
+
+	// Re-apply forcing the changes should succeed
+	_, err = client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Name("deployment").
+		Param("fieldManager", "apply_test").
+		Param("force", "true").
+		Body(validDeployment).
+		Do(context.TODO()).
+		Get()
+	if err != nil {
+		t.Fatalf("Error updating deployment: %v ", err)
+	}
+
+	deployment, err = client.AppsV1().Deployments("default").Get(context.TODO(), "deployment", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to retrieve object: %v", err)
+	}
+
+	if *deployment.Spec.Replicas != 1 {
+		t.Fatalf("Expected replicas to be 1, but got: %d", *deployment.Spec.Replicas)
+	}
+
+	assertReplicasOwnership(t, (*deployment).GetManagedFields(), "apply_test")
+
+	// Replace scale object
+	_, err = client.CoreV1().RESTClient().
+		Put().
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		SubResource("scale").
+		Name("deployment").
+		Param("fieldManager", "replace_test").
+		Body([]byte(`{"kind":"Scale","apiVersion":"autoscaling/v1","metadata":{"name":"deployment","namespace":"default"},"spec":{"replicas":7}}`)).
+		Do(context.TODO()).
+		Get()
+	if err != nil {
+		t.Fatalf("Error updating deployment: %v ", err)
+	}
+
+	deployment, err = client.AppsV1().Deployments("default").Get(context.TODO(), "deployment", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to retrieve object: %v", err)
+	}
+
+	if *deployment.Spec.Replicas != 7 {
+		t.Fatalf("Expected to replicas to be 7, but got: %d", *deployment.Spec.Replicas)
+	}
+
+	assertReplicasOwnership(t, (*deployment).GetManagedFields(), "replace_test")
+}
+
+func assertReplicasOwnership(t *testing.T, managedFields []metav1.ManagedFieldsEntry, fieldManager string) {
```

Comment:
I'm really not a big fan of this here. Other suggestion:
1. Check that there is an entry for the manager used in the scale operation,
2. Check for conflicts on changes to that field

Reply:
I am not sure I fully understood.

> Check that there is an entry for the manager used in the scale operation,

This is done [here](https://github.com/kubernetes/kubernetes/pull/95921/files#diff-b629323ef67f91837456deae39526a0e81aa0a5397465ff546e6845211afb13dR2992-R2994) unless I missed something. `fieldManager` is the manager used in the scale operation, see for instance [here](https://github.com/kubernetes/kubernetes/pull/95921/files#diff-b629323ef67f91837456deae39526a0e81aa0a5397465ff546e6845211afb13dR2896).

> Check for conflicts on changes to that field

[This](https://github.com/kubernetes/kubernetes/pull/95921/files#diff-b629323ef67f91837456deae39526a0e81aa0a5397465ff546e6845211afb13dR2898-R2910) was my attempt to check for conflict, since the checks happens right after applying I thought in made sense to check it outside an helper.

Reply:
I couldn't find an easy we have a way to check the that `replicas` field is owned by a field manager (and not others) without decoding. I am open to suggestions though.

I see your point of not checking internal structures, but we don't have unit tests for this feature and I would feel more confident having a more strict integration tests. As we discussed, we could definitely refactor this bit into a clearer framework.

WDYT?